### PR TITLE
Add Org-roam support to key-at-point

### PR DESCRIPTION
**Pull request by zaeph**
Not sure whether we should make this a function, as we did elsewhere.

------
I couldn't find where we'd discussed this, so I can't link back, sadly.

### CODE REVIEW

**Comment by zaeph** [comment]
Frankly, I think we can do without the `funcall` to a customisable function, here.  The `require` with silent failure should be enough, even though Org-roam isn't a dependency.

**Comment by zaeph** [comment]
Went ahead and did it in ee5870a57d6a457ad9d58b2a9208d5f0b364123d because, well, why not.

File: bibtex-completion.el [hunk 1]
```diff
@@ -1537,41 +1537,64 @@ BibTeX files.  If this fails, return nil."
...
+                                    (if (boundp 'org-bibtex-key-property)
+                                        org-bibtex-key-property
+                                      "CUSTOM_ID")
+                                    t))
+           ;; KEY may be the empty string the the property is
+           ;; present but has no value
+           (> (length key) 0)
+           key))))
+
+(defvar bibtex-completion-key-at-point-functions
```
Comment (zaeph) [inline bibtex-completion.el:1583]: Went for a `defvar` instead of a `defcustom` because it's geared towards packages rather than users.

**Comment by tmalsburg** [comment]
I've added some minor comments.  Looks good otherwise.

**Comment by zaeph** [comment]
Back from a writing hiatus, sorry for the delay.

> I've added some minor comments.

Are you sure?  I don't see them.

**Comment by tmalsburg** [comment]
I didn't add them in the code but here (see above).

**Comment by zaeph** [comment]
Are you sure you've validated those comments?  They don't appear on my end.

**Comment by tmalsburg** [comment]
Do you seem them now?  Not super familiar with the review features.

**Comment by zaeph** [comment]
Yes, they do!

I'll have a look later today.

File: bibtex-completion.el [hunk 2]
```diff
@@ -1537,41 +1537,64 @@ BibTeX files.  If this fails, return nil."
...
+This function can be used by `bibtex-completion-key-at-point' to
+find the key of the BibTeX entry at point in a BibTeX-mode
+buffer."
+  (when (eq major-mode 'bibtex-mode)
+    (save-excursion
+      (bibtex-beginning-of-entry)
+      (and (looking-at bibtex-entry-maybe-empty-head)
+           (bibtex-key-in-head)))))
+
+(defun bibtex-completion-get-key-reftex ()
```
[the remark this answers is not in the document]
Comment (zaeph) [reply bibtex-completion.el:1551]: Addressed in 543e7f56d76b914a448267c532cb10d1d9b746e9.

File: bibtex-completion.el [hunk 3]
```diff
@@ -1537,41 +1537,64 @@ BibTeX files.  If this fails, return nil."
...
+      (let ((macro (reftex-what-macro 1)))
+        (and (stringp (car macro))
+             (string-match "\\`\\\\cite\\|cite\\'" (car macro))
+             ;; allow '_' in citekeys
+             (let ((temp-syn-table (make-syntax-table)))
+               (modify-syntax-entry ?_ "_" temp-syn-table)
+               (with-syntax-table temp-syn-table
+                 (thing-at-point 'symbol))))))))
+
+(defun bibtex-completion-get-key-org ()
```
[the remark this answers is not in the document]
Comment (zaeph) [reply bibtex-completion.el:1567]: Addressed in 543e7f56d76b914a448267c532cb10d1d9b746e9.

File: bibtex-completion.el [hunk 4]
```diff
@@ -1537,41 +1537,64 @@ BibTeX files.  If this fails, return nil."
...
+    (save-excursion
+      (bibtex-beginning-of-entry)
+      (and (looking-at bibtex-entry-maybe-empty-head)
+           (bibtex-key-in-head)))))
+
+(defun bibtex-completion-get-key-reftex ()
+  "Return the key of the BibTeX entry at point, nil otherwise.
+This function can be used by `bibtex-completion-key-at-point' to
+find the key of the BibTeX entry at point in a LaTeX buffer."
+  (when (require 'reftex-parse nil t)
```
[the remark this answers is not in the document]
Comment (zaeph) [reply bibtex-completion.el:1555]: Addressed in 59609f6e87d946d244f7fe41d829dd951270cd79.

**Comment by zaeph** [comment]
@tmalsburg Friendly reminder that this PR is ready for merging. :)

**Comment by tmalsburg** [comment]
Uh, sorry for the delay. It's been hectic and this week I'm (finally) on a shot vacation.  Will revisit this next week.  Thanks for the reminder.

**Comment by zaeph** [comment]
No worries whatsoever: take your time!

**Comment by tmalsburg** [comment]
Didn't have time to read this PR closely again, but I don't want to hold this back for too long and trust that the PR is in good shape.

**Pull request merged**